cfgmgr/buffermgr.cpp

```diff
@@ -112,11 +112,27 @@ Create/update two tables: profile (in m_cfgBufferProfileTable) and port buffer (
         }
     }
 */
-task_process_status BufferMgr::doSpeedUpdateTask(string port, string speed)
+task_process_status BufferMgr::doPortTableUpdateTask(string port, string speed, bool admin_up)
 {
     vector<FieldValueTuple> fvVector;
     string cable;
 
+    string buffer_pg_key = port + m_cfgBufferPgTable.getTableNameSeparator() + LOSSLESS_PGS;
+
+    m_cfgBufferPgTable.get(buffer_pg_key, fvVector);
+
+    if (!admin_up)
+    {
+        // Remove the entry in BUFFER_PG table if any
+        if (!fvVector.empty())
+        {
+            SWSS_LOG_NOTICE("Removing PG %s from port %s which is administrative down", buffer_pg_key.c_str(), port.c_str());
+            m_cfgBufferPgTable.del(buffer_pg_key);
```

Comment:
Please make this change platform specific. This will cause issues on other vendor asics

Reply:
As discussed, there is no vendor info passed to `buffermgrd` for now. I'll open a PR to sonic-buildimage to have an env variable passed to buffer manager and then fix this one.

Reply:
Fixed.
An environment variable `ASIC_VENDOR` is loaded at the beginning of the daemon and will be tested here. The admin-down will be handled on Mellanox platform only.
It depends on another sonic-buildimage PR which loads `ASIC_VENDOR` to swss docker and will come soon.